### DISCUSSION TITLE
Providers with `yield`

### DISCUSSION THREAD

**Discussion by provinzkraut** [litestar-org/litestar#678] [Ideas]
On a [reddit post](https://www.reddit.com/r/Python/comments/yc7rvf/comment/itl41oy) someone brought up supporting generator based providers, so that cleanup code could be run after the request.

**Comment by Goldziher** [comment]
i think this is a pretty neat idea, i dont see a reason we shouldnt implement this support - its like pytest. What do you think?

**Reply by provinzkraut** [reply]
I think we should support this. It's an established pattern in python and shouldn't be too difficult to implement.

**Comment by provinzkraut** [comment]
Implemented in #856
